packages/cli/src/commands/blob/put.ts

```diff
@@ -53,49 +46,69 @@ export default async function put(
     '--force': force,
   } = flags;
 
-  telemetryClient.trackCliArgumentPathToFile(filePath);
+  // Only track file path if one was provided
+  if (filePath) {
+    telemetryClient.trackCliArgumentPathToFile(filePath);
+  }
   telemetryClient.trackCliFlagAddRandomSuffix(addRandomSuffix);
   telemetryClient.trackCliOptionPathname(pathnameFlag);
   telemetryClient.trackCliFlagMultipart(multipart);
   telemetryClient.trackCliOptionContentType(contentType);
   telemetryClient.trackCliOptionCacheControlMaxAge(cacheControlMaxAge);
   telemetryClient.trackCliFlagForce(force);
 
-  let putBody: NodeJS.ReadableStream;
+  // ReadableStream works for both stdin and ReadStream
+  let putBody: ReadableStream;
   let pathname: string;
 
-  try {
-    const stats = statSync(filePath);
-    const isFile = stats.isFile();
-
-    if (isFile) {
-      // we first open the file so we can handle errors with promises
-      const file = await open(filePath, 'r');
-      putBody = file.createReadStream();
-      putBody.once('error', err => {
-        output.error(`Error while reading file: ${err}`);
-      });
-      pathname = pathnameFlag ?? basename(filePath);
-    } else {
-      output.error('Path to upload is not a file');
+  if (!filePath) {
+    // Check if stdin is a TTY (user is typing directly in terminal)
+    // isTTY is undefined when using as cat file.txt | vercel blob put
+    if (client.stdin.isTTY) {
+      output.error(
+        `Missing input. Usage: ${chalk.cyan(
+          `${getCommandName('blob put <file>')}`
+        )} or pipe data: ${chalk.cyan('cat file.txt | vercel blob put')}`
+      );
       return 1;
     }
-  } catch (err) {
-    output.debug(`Error reading file: ${err}`);
 
-    if (isErrnoException(err)) {
-      output.error(`File doesn't exist at '${filePath}'`);
+    // Reading from stdin
+    putBody = process.stdin;
+    pathname = pathnameFlag ?? `${new Date().toISOString()}-from-stdin`;
```

Comment:
should we default here. Seems weird to get a random file name. I personally would prefer an error so that I can use the command correctly

Reply:
I wondered the same but then it makes the --pathname in one case required in another case not required. Given this usage is edge case it's probably ok? And we can ensure in the docs we show it with --pathname